tests/model/test_modeler.py

```diff
@@ -0,0 +1,114 @@
+import os
+from typing import Any, Dict
+import tempfile
+import pandas as pd
+from sklearn import datasets
+from sklearn.linear_model import LogisticRegression
+from sklearn.preprocessing import StandardScaler
+from sklearn.metrics import r2_score
+from h1st.model.model import Model
+from h1st.model.ml_modeler import MLModeler
+from h1st.model.ml_model import MLModel
+
+class MyMLModeler(MLModeler):
+    def __init__(self):
+        self.stats = {}
+        self.example_test_data_ratio = 0.2
+
+    def load_data(self) -> Dict:
+        df_raw = datasets.load_iris(as_frame=True).frame
+        return self.generate_training_data({'df_raw': df_raw} )
+    
+    def preprocess(self, data):
+        self.stats['scaler'] = StandardScaler()
+        return self.stats['scaler'].fit_transform(data) 
+    
+    def generate_training_data(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        df_raw = data['df_raw']
+        df_raw.columns = ['sepal_length','sepal_width','petal_length','petal_width', 'species']
+        
+        self.stats['targets'] = sorted(df_raw['species'].unique())
+        self.stats['targets_dict'] = {k: v for v, k in enumerate(self.stats['targets'])}
+        df_raw['species'] = df_raw['species'].apply(lambda x: self.stats['targets_dict'][x])
+
+        # Shuffle all the df_raw
+        df_raw = df_raw.sample(frac=1, random_state=5).reset_index(drop=True)
+        
+        # Preprocess data
+        df_raw.loc[:, 'sepal_length':'petal_width'] = self.preprocess(
+            df_raw.loc[:, 'sepal_length':'petal_width'])
+
+        # Split to training and testing data
+        n = df_raw.shape[0]
+        n_test = int(n * self.example_test_data_ratio)
+        training_data = df_raw.iloc[n_test:, :].reset_index(drop=True)
+        test_data = df_raw.iloc[:n_test, :].reset_index(drop=True)
+
+        # Split the data to features and labels
+        train_data_x = training_data.loc[:, 'sepal_length':'petal_width']
+        train_data_y = training_data['species']
+        test_data_x = test_data.loc[:, 'sepal_length':'petal_width']
+        test_data_y = test_data['species']
+
+        # When returning many variables, it is a good practice to give them names:
+        return {
+            'train_x':train_data_x,
+            'train_y':train_data_y,
+            'test_x':test_data_x,
+            'test_y':test_data_y,
+        }
+
+    def train(self, data: Dict[str, Any]) -> Any:
+        X, y = data['train_x'], data['train_y']
+        model = LogisticRegression(random_state=0)
+        model.fit(X, y)
+        return model
+    
+    # TODO: need to check model instance type
+    def evaluate(self, data: Dict, model: Model) -> Dict:
```

Comment:
1. I think, model should be MLModel instead of Model. 
2. In evaluate() function declaration, it returns Dict. But inside the function, it doesn't return anything. 
3. I think it is better to return evaluation in dictionary. => it is related to the comments I made in MLModeler.build function.

Reply:
1. You have to be able to evaluate a non-MLModel, right?
2, 3. Agree, let me update the code.

Reply:
1. In modeler yes, but inside MLModeler, we should evaluate MLModel, right ?

Reply:
OKay!!! Then we need to update the MLModel too, right?